libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayRouteMapper.kt

```diff
@@ -0,0 +1,93 @@
+package com.mapbox.navigation.core.replay.history
+
+import android.location.Location
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.geojson.LineString
+import com.mapbox.geojson.Point
+import com.mapbox.turf.TurfConstants
+import com.mapbox.turf.TurfMeasurement
+import java.util.Collections
+
+class ReplayRouteMapper(
```

Comment:
please add docs for public classes and methods

Reply:
We're missing multi-leg support to have parity with current `ReplayRouteLocationEngine` / `ReplayRouteLocationConverter` 👀 

https://github.com/mapbox/mapbox-navigation-android/blob/3044db38b6d26cd42fe69d7fc3544e1f81add612/libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/route/ReplayRouteLocationEngine.kt#L116 https://github.com/mapbox/mapbox-navigation-android/blob/3044db38b6d26cd42fe69d7fc3544e1f81add612/libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/route/ReplayRouteLocationConverter.kt#L21-L24

Also was wondering if we could re-use some code across `replay.history` and `replay.route` until `route` is replaced completely by `history`. What do you think?

Reply:
good catch, where are some examples of multi-leg? 

I'll see what it does when adding some tests

Reply:
Wondering if we should do this now or after we complete this
https://github.com/mapbox/navigation-sdks/issues/315

Reply:
Probably after :sweat_smile:

Reply:
Multiple initiatives to move forward waypoint support
https://github.com/mapbox/mapbox-navigation-android/pull/2786
https://github.com/mapbox/mapbox-navigation-android/pull/2787

making it so we can iterate on these things separately

Reply:
also docs added 🎈